engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/NadelDeepRenameTransform.kt

```diff
@@ -58,7 +61,7 @@ internal class NadelDeepRenameTransform : NadelTransform<NadelDeepRenameTransfor
          * }
          * ```
          */
-        val instructions: Map<FieldCoordinates, NadelDeepRenameFieldInstruction>,
+        val instructionsByObjectTypeNames: Map<GraphQLObjectTypeName, NadelDeepRenameFieldInstruction>,
```

Comment:
Changed the key to be a `String` aliased as `GraphQLObjectTypeName`. Having the key as `FieldCoordinates` never really made sense because the only thing different about any of these coordinates was the type. The field MUST have been the same.

You'll notice this weird behavior documented in some of the deleted code in this PR

```kotlin
    // NOTE: the given instructions must have the same field name, just differing type name
    // Otherwise this function doesn't make sense
    val fieldName = instructions.keys.first().fieldName

    instructions[makeFieldCoordinates(overallTypeName, fieldName)]
```

Whereas now the code is just

```kotlin
    instructions[overallTypeName]
```